packages/core/src/shouldIntercept.ts

```diff
@@ -1,4 +1,13 @@
-export default function shouldIntercept(event: MouseEvent | KeyboardEvent): boolean {
+// The actual event passed to this function could be a native JavaScript event
+// or a React synthetic event, so we are picking just the keys needed here (that
+// are present in both types).
+
+export default function shouldIntercept(
+  event: Pick<
+    MouseEvent,
+    'altKey' | 'ctrlKey' | 'defaultPrevented' | 'target' | 'currentTarget' | 'metaKey' | 'shiftKey' | 'button'
+  >,
```

Comment:
There might be a more elegant way to do this? I didn't want to introduce a dependency on React type definitions in the core package (to make this a union type), so figured the next best thing was to narrow this down to accommodate both native and synthetic event shapes. 

I confirmed that TypeScript _will_ fail if this is type is just a `MouseEvent`, since `React.MouseEvent` contains some different stuff.